pgxn/neon/logical_replication_monitor.c

```diff
@@ -252,31 +253,37 @@ LogicalSlotsMonitorMain(Datum main_arg)
 				ReplicationSlot *s = &ReplicationSlotCtl->replication_slots[i];
 				XLogRecPtr	restart_lsn;
 
-				/* find the name */
 				LWLockAcquire(ReplicationSlotControlLock, LW_SHARED);
```

Comment:
Should we not acquire the lock before getting the slot out of the array?

Reply:
I am not sure to whom this question is, but getting address of slot before acquiring locks is ok: this array can not be reallocated.